### PR TITLE
Optimise "getLanguage()" method #86

### LINKED ISSUE #86: Optimise  "getLanguage()" method
Related discussion https://cq-dev.slack.com/archives/C9KD0TT6G/p1678687361360909?thread_ts=1678685048.805829&cid=C9KD0TT6G

### PR DESCRIPTION
**Pull request by bosschaert**
Fixes #86

Test URLs:
- Before:  https://main--nedbank--hlxsites.hlx.page/
- After: https://issue-86--nedbank--hlxsites.hlx.page/

### CODE REVIEW

File: scripts/scripts.js [hunk 1]
```diff
@@ -23,22 +23,24 @@ export const LANG = {
 
 let language;
 
-export function getLanguage() {
+export function getLanguageFromPath(pathname) {
```
Comment (bosschaert) [inline scripts/scripts.js:26]: I split the main logic off to a separate function to make unit testing easier (no dependency on `window.location.pathname`)

**Comment by bosschaert** [comment]
Note that the test failures already existed in the codebase, but I did add a new unit test for my changes.

**Review by sdmcraft — APPROVED** [review]
Thanks @bosschaert !

File: scripts/scripts.js [hunk 2]
```diff
@@ -23,22 +23,24 @@ export const LANG = {
...
   if (language) return language;
```
Comment (tripodsan) [inline scripts/scripts.js:27]: this is not really efficient, if language remains not found (empty string).

File: scripts/scripts.js [hunk 3]
```diff
@@ -23,22 +23,24 @@ export const LANG = {
...
-  const segs = window.location.pathname.split('/');
-  if (segs && segs.length > 0) {
-    // eslint-disable-next-line no-restricted-syntax
-    for (const [, value] of Object.entries(LANG)) {
-      if (value === segs[1]) {
-        language = value;
-        break;
-      }
+  const segs = pathname.split('/');
+  if (segs.length > 0) {
```
Comment (tripodsan) [inline scripts/scripts.js:31]: `> 1` !

File: scripts/scripts.js [hunk 4]
```diff
@@ -23,22 +23,24 @@ export const LANG = {
...
+    const l = LANG[segs[1].toUpperCase()];
+    if (l) {
+      language = l;
```
Comment (tripodsan) [inline scripts/scripts.js:34]: why do you need the uppercase langs at all? you could just have a set of all valid langs.

**Pull request merged**